### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR adds some performance improvements to edit form modals and wraps callbacks in `useCallback`.

See #2600

### CODE REVIEW

File: assets/src/application/ui/forms/espressoForm/EspressoForm.tsx [hunk 1]
```diff
@@ -10,18 +10,15 @@ import './styles.scss';
 
 const EspressoForm: React.FC<EspressoFormProps> = ({ onSubmit, mutators, layout, ...rest }) => {
 	const context = { layout };
+	const formMutators = {
```
Comment (nicolad) [inline assets/src/application/ui/forms/espressoForm/EspressoForm.tsx:13]: :+1:

File: assets/src/application/ui/forms/espressoForm/EspressoForm.tsx [hunk 2]
```diff
@@ -10,18 +10,15 @@ import './styles.scss';
...
-				onSubmit={onSubmit}
-				render={FormRenderer}
-				mutators={{
-					...arrayMutators,
-					...mutators,
-					updateFieldValue,
-				}}
-				{...rest}
-			/>
+			<ReactFinalForm component={FormRenderer} onSubmit={onSubmit} mutators={formMutators} {...rest} />
```
Comment (nicolad) [inline assets/src/application/ui/forms/espressoForm/EspressoForm.tsx:21]: is it safe to pass the rest of the props as last argument here?

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useOnCreateDatetime.ts [hunk 3]
```diff
@@ -9,34 +11,37 @@ const useOnCreateDatetime = (): DatetimeMutationCallbackFn => {
...
-				relation: 'tickets',
-				relationIds: tickets,
-			});
-			tickets.forEach((entityId: string) => {
-				addRelation({
-					entity: 'tickets',
-					entityId,
-					relation: 'datetimes',
-					relationId: datetimeId,
+	const onCreateDatetime = useCallback(
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useOnCreateDatetime.ts:14]: I also added useCallback :)
in this PR: https://github.com/eventespresso/event-espresso-core/pull/2862

File: assets/src/domain/eventEditor/services/apollo/queries/tickets/useTicketQueryOptions.ts [hunk 4]
```diff
@@ -26,7 +27,8 @@ const useTicketQueryOptions = (datetimeIn: EntityId[] = []): DatetimesQueryOptio
 		},
 	};
 
-	return options;
+	const datetimeInStr = datetimeIn.join(':');
+	return useMemo(() => options, [datetimeInStr]);
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/queries/tickets/useTicketQueryOptions.ts:31]: I've created a helper, will extend with this approach as well, after the merge:

https://github.com/eventespresso/event-espresso-core/pull/2862/files#diff-2368976113d2a8bc6cdbeb64364abdb6

File: assets/src/application/ui/forms/espressoForm/EspressoForm.tsx (hunk 2 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/forms/espressoForm/EspressoForm.tsx:21]: Yes, consumer can have full control

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/components/TicketAssignmentsManagerModal/index.tsx [hunk 5]
```diff
@@ -7,27 +7,25 @@ import CloseModalButton from './buttons/CloseModal';
...
+import { WithContextProps } from '../../context/types';
 
 import '../styles.scss';
 
-const TicketAssignmentsManagerModal: React.FC = () => {
-	const {
-		dataState: { hasOrphanEntities },
-		onCloseModal,
-		title,
-	} = useTAMContext();
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/components/TicketAssignmentsManagerModal/index.tsx:19]: Using context here caused re-renders.

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/components/TicketAssignmentsManagerModal/index.tsx:19]: would be nice to document this conclusion somewhere, along with that comment you posted yesterday: https://github.com/facebook/react/issues/15156#issuecomment-474590693

File: assets/src/application/services/utilities/memo/reactFinalFormState.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { FormRenderProps } from 'react-final-form';
 
 const reactFinalFormState = (prevProps: FormRenderProps, nextProps: FormRenderProps): boolean => {
-	const prevValue = JSON.stringify(prevProps.form.getState());
-	const nextValue = JSON.stringify(nextProps.form.getState());
+	const prevValue = JSON.stringify(prevProps.form?.getState());
+	const nextValue = JSON.stringify(nextProps.form?.getState());
```
Comment (nicolad) [inline assets/src/application/services/utilities/memo/reactFinalFormState.ts]: agree, better safe :)

**Pull request merged**